src/environment.ts

```diff
@@ -145,117 +105,53 @@ export class SecureEnvironment implements MembraneBroker {
-                                // an example of this is circular window.window ref
-                                console.info('circular: ',  key, blueValue, blueDescriptor);
-                            }
-                        }
-                    } else if (hasOwnProperty(redDescriptor, 'get')) {
-                        // internationally ignoring the case of (typeof document.all === 'undefined') because
-                        // it is specified as configurable, you never get one of those exotic objects in this branch
-                        if (typeof redPropertyValue === 'function' || typeof redPropertyValue === 'object') {
-                            if (redPropertyValue === redValue[key]) {
-                                // this is the case for window.document which is identity preserving getter
-                                // const blueDescriptorValue = blueValue[key];
-                                // this.setRefMapEntries(redDescriptorValue, blueDescriptorValue);
-                                // this.installDescriptors(redDescriptorValue, blueDescriptorValue, getOwnPropertyDescriptors(blueDescriptorValue));
-                                console.error('need remapping: ', key, blueValue, blueDescriptor);
-                                if (ReflectIsExtensible(redPropertyValue)) {
-                                    // remapping proto chain
-                                    // ReflectSetPrototypeOf(redDescriptorValue, this.getRedValue(ReflectGetPrototypeOf(redDescriptorValue)));
-                                    console.error('needs prototype remapping: ', key, blueValue);
-                                } else {
-                                    console.error('leaking prototype: ',  key, blueValue, blueDescriptor);
-                                }
-                            } else {
-                                console.error('leaking a getter returning values without identity: ', key, blueValue, blueDescriptor);
-                            }
-                        } else {
-                            console.error('skipping: ', key, blueValue, blueDescriptor);
-                        }
-                    }
-                } else {
-                    ReflectDefineProperty(redValue, key, blueDescriptor);
+                if (isFunction(blueDescriptor.set)) {
+                    redDescriptor.set = function(v: RedValue): void {
+                        // if a global setter is invoke, the value will be use as it is as the result of the getter operation
+                        currentBlueGetter = () => v;
+                    };
                 }
             }
+            ReflectDefineProperty(redValue, key, redDescriptor);
         }
     }
 }
+
+function canRedPropertyBeTamed(redValue: RedValue, key: PropertyKey): boolean {
+    const redDescriptor = ReflectGetOwnPropertyDescriptor(redValue, key);
+    // TODO: what about writable - non-configurable?
```

Comment:
although in browsers we never get a non-configurable writable as a global descriptor, we might want to get this resolved at some point.

Reply:
Good abstraction! But writable non-configurable is semantically okay, no?

Reply:
yes, in theory, we could diff the current descriptor against the new descriptor, we could passed as argument, and determine that only the value is changing, in which case it can be tamed since it is writable.